extensions/wikia/ContentReview/ContentReviewHelper.php

```diff
@@ -112,25 +117,48 @@ public static function isStatusAwaiting( $status ) {
 	}
 
 	public function isDiffPageInReviewProcess( $wikiId, $pageId, $diff ) {
+		global $wgRequest;
+
+		/**
+		 * Do not hit database if there is a URL parameter that indicates that a user
+		 * came directly from Special:ContentReview.
+		 */
+		if ( $wgRequest->getInt( self::CONTENT_REVIEW_URL_PARAM ) === 1 ) {
+			return true;
+		}
 
 		$reviewModel = new ReviewModel();
 		$reviewData = $reviewModel->getReviewedContent( $wikiId, $pageId, ReviewModel::CONTENT_REVIEW_STATUS_IN_REVIEW );
 
-		if ( !empty( $reviewData ) && (int)$reviewData['revision_id'] === $diff ) {
-			return true;
-		}
-		return false;
+		return ( !empty( $reviewData ) && (int) $reviewData['revision_id'] === $diff );
 	}
 
 	public function hasPageApprovedId( $wikiId, $pageId, $oldid ) {
-
 		$currentModel = new CurrentRevisionModel();
 		$currentData = $currentModel->getLatestReviewedRevision( $wikiId, $pageId );
 
-		if ( !empty( $currentData ) && (int)$currentData['revision_id'] === $oldid ) {
-			return true;
-		}
-		return false;
+		return ( !empty( $currentData ) && (int)$currentData['revision_id'] === $oldid );
+	}
+
+	public function getToolbarTemplate() {
+		global $wgCityId, $wgTitle;
+
+		return \MustacheService::getInstance()->render(
+			self::CONTENT_REVIEW_TOOLBAR_TEMPLATE_PATH,
+			[
+				'toolbarTitle' => wfMessage( 'content-review-diff-toolbar-title' )->plain(),
+				'wikiId' => $wgCityId,
+				'pageId' => $wgTitle->getArticleID(),
+				'approveStatus' => ReviewModel::CONTENT_REVIEW_STATUS_APPROVED,
+				'buttonApproveText' => wfMessage( 'content-review-diff-approve' )->plain(),
+				'rejectStatus' => ReviewModel::CONTENT_REVIEW_STATUS_REJECTED,
+				'buttonRejectText' => wfMessage( 'content-review-diff-reject' )->plain(),
+				'talkpageUrl' => $this->prepareProvideFeedbackLink( $wgTitle ),
+				'talkpageLinkText' => wfMessage( 'content-review-diff-toolbar-talkpage' )->plain(),
+				'guidelinesUrl' => wfMessage( 'content-review-diff-toolbar-guidelines-url' )->plain(),
```

Comment:
In general, if it's a URL able to be set by a MW message, I think it's preferable to use `->useDatabase( false )`. While admins can only edit whitelisted MW pages right now, if they were able to edit this again, it could be set to any URL.